modules/external-auth-configuring.adoc

```diff
@@ -0,0 +1,190 @@
+    - clientID: console-test <9>
+      clientSecret:
+        name: console-secret <10>
+      componentName: console
+      componentNamespace: openshift-console
+  type: OIDC <11>
+  webhookTokenAuthenticator: null <12>
+----
+<1> The OIDC provider configuration. Currently, only one OIDC provider configuration is allowed.
+<2> The name of the claim to construct group names for the cluster identity.
+<3> The name of the claim to construct usernames for the cluster identity.
+<4> The list of audience IDs that this authentication provider issues tokens for.
+<5> The name of the config map that contains the `ca-bundle.crt` key. If unset, system trust is used instead.
+<6> The URL for the token issuer.
+<7> The name for external OIDC provider.
+<8> The client ID that your provider uses for the {oc-first}.
+<9> The client ID that your provider uses for the {product-title} web console.
+<10> The name of the secret that stores the secret value for the console client.
+<11> Must be set to `OIDC` to indicate to use an external OIDC identity provider.
+<12> Must be set to `null` when `type` is set to `OIDC`.
++
+For more details on all available parameters, see "OIDC provider configuration parameters".
+
+. Exit and save the changes to apply the new configuration.
+
+.Verification
+
+. Wait for the cluster to roll out new revisions to all nodes.
+
+.. Check the Kubernetes API server Operator status by running the following command:
++
+[source,terminal]
+----
+$ oc get co kube-apiserver
+----
++
+.Example output
+[source,terminal]
+----
+NAME             VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
+kube-apiserver   4.19.0    True        True          False      85m     NodeInstallerProgressing: 2 node are at revision 8; 1 node is at revision 10
+----
++
+The message in the preceding example shows that one node has progressed to the new revision and two nodes have not yet updated. It can take 20 minutes or more to roll out the new revision to all nodes, depending on the size of your cluster.
+
+.. To troubleshoot any issues, you can also check the Cluster Authentication Operator and `kube-apiserver` pod logs for errors.
+
+. Verify that you can log in to the {oc-first} by authenticating with Keycloak:
+
+.. Log in by running the following command:
++
+[source,terminal]
+----
+$ oc login --exec-plugin=oc-oidc \// <1>
+    --issuer-url=https://keycloak-keycloak.apps.example.com/realms/master \// <2>
+    --client-id=oc-cli-test \// <3>
+    --extra-scopes=email --callback-port=8080 \
+    --oidc-certificate-authority my-directory/ca-bundle.crt <4>
+----
+<1> Specify `oc-oidc` as the exec plugin type. Only a value of `oc-oidc` is allowed.
+<2> Specify the issuer URL for your Keycloak provider.
+<3> Specify client ID for the {oc-first}.
+<4> Specify the path to the `ca-bundle.crt` file on your local machine.
++
+.Example output
+[source,terminal]
+----
+WARNING: Using insecure TLS client config. Setting this option is not supported!
```

Comment:
@bergerhoffer This is not necessary. If the kubeconfig already has certificate-authority-data of current cluster, this won't show.

Reply:
Great thanks I'll remove that line.